### PR TITLE
Updated Unknown Error within RFSA/RFSG example

### PR DESCRIPTION
**Pull request by IPetersNI**
### What does this Pull Request accomplish?

- Handles the UNKNOWN status code for arguments within the command line interface.
- Makes UNKNOWN error easier to identify and change.

### Why should this Pull Request be merged?

[Task 2125465](https://dev.azure.com/ni/DevCentral/_workitems/edit/2125465): Update examples (RFSA/RFSG) to handle new errors

### What testing has been done?
Test were done with both valid and invalid inputs that resulted in the error for all 4 files for RFSA/RFSG

### CODE REVIEW

File: examples/nirfsa/getting-started-iq.py [hunk 1]
```diff
@@ -148,11 +148,14 @@ def raise_if_initialization_error(response):
     error_message = rpc_error.details()
     if rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
         error_message = f"Failed to connect to server on {SERVER_ADDRESS}:{SERVER_PORT}"
+
```
Comment (dmondrik) [inline examples/nirfsa/getting-started-iq.py:151]: You'll find that extra whitespace is not optional with the formatter.

File: examples/nirfsa/getting-started-iq.py [hunk 2]
```diff
@@ -148,11 +148,14 @@ def raise_if_initialization_error(response):
...
     elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
         error_message = (
             "The operation is not implemented or is not supported/enabled in this service"
         )
+    elif rpc_error.code() == grpc.StatusCode.UNKNOWN:
+        error_message = ("Invalid Input: " + RESOURCE)
```
Comment (dmondrik) [inline examples/nirfsa/getting-started-iq.py:157]: You need to follow the pattern for what I did [in my PR](https://github.com/ni/grpc-device/pull/709/files). See lines 213-218 specifically.

**Comment by dmondrik** [comment]
You also need to remove all the `raise_if...error` calls because there's no longer a need to check for errors after the call.

File: examples/nirfsa/getting-started-iq.py (hunk 1 above)
Comment (astarche) [reply examples/nirfsa/getting-started-iq.py:151]: Consider using black locally. We have some instructions here: https://github.com/ni/grpc-device/wiki/Development-Tips-and-Tricks#black-format-on-save

**Review by reckenro — APPROVED** [review]

File: examples/nirfsa/getting-started-iq.py [hunk 3]
```diff
@@ -60,78 +61,53 @@
...
-        raise RuntimeError(f"Error: {response.error_message or response.status}")
+def check_for_warning(response, vi):
+    """Print to console if the status indicates a warning."""
     if response.status > 0:
-        sys.stderr.write(f"Warning: {response.error_message or response.status}\n")
-    return response
+        warning_message = client.ErrorMessage(
+            nirfsa_types.ErrorMessageRequest(vi=vi, error_code=response.status)
+        )
+        sys.stderr.write(f"{warning_message}\nWarning status: {response.status}\n")
```
Comment (dmondrik) [inline examples/nirfsa/getting-started-iq.py:70]: I know you copied this, but I think it's wrong in my original.

The data type of `warning_message` is not a string, it's `ErrorMessageResponse`. Which means what we print to the console should be `warning_message.error_message`. Right @reckenro ?

@danestull if your PR isn't submitted yet, can you include this fix in yours too?

File: examples/nirfsa/getting-started-spectrum.py [hunk 4]
```diff
@@ -64,86 +65,60 @@
...
-            )
+    read_response = client.ReadPowerSpectrumF64(
+        nirfsa_types.ReadPowerSpectrumF64Request(
+            vi=vi,
+            timeout=10.0,
+            data_array_size=spectral_lines_response.number_of_spectral_lines,
         )
     )
-
+    check_for_warning(read_response, vi)
```
Comment (dmondrik) [inline examples/nirfsa/getting-started-spectrum.py:118]: There was a blank line between the read and the comment block. You should keep 1 blank line after the warning check.

**Review by dmondrik — APPROVED** [review]
Once you apply the suggested changes, feel free to submit it.

File: examples/nirfsa/getting-started-iq.py (hunk 3 above)
Comment (reckenro) [reply examples/nirfsa/getting-started-iq.py:70]: Oh yes, good catch. I missed this in that initial PR, it should be `warning_message.error_message` here like you said.

File: examples/nirfsg/getting-started-single-tone-generation.py [hunk 5]
```diff
@@ -56,20 +57,13 @@
...
-        raise RuntimeError(f"Error: {response.error_message or response.status}")
+def check_for_warning(response, vi):
+    """Print to console if the status indicates a warning."""
     if response.status > 0:
-        sys.stderr.write(f"Warning: {response.error_message or response.status}\n")
-    return response
+        warning_message = client.ErrorMessage(
+            nirfsg_types.ErrorMessageRequest(vi=vi, error_code=response.status)
+        )
+        sys.stderr.write(f"{warning_message}\nWarning status: {response.status}\n")
```
Comment (dmondrik) [inline examples/nirfsg/getting-started-single-tone-generation.py:66]: My comment about the type applied to all of them.

**Review by dmondrik — APPROVED** [review]

**Review by astarche — APPROVED** [review]

**Pull request merged**